NEWS

```diff
@@ -1,3 +1,95 @@
+libxkbcommon 1.7.0 - XXX
+==================
+
+TODO: date
+
+API
+---
+
+### New
+
+- Added early detection of invalid encodings and BOM detection for keymaps, rules
+  & Compose. Also added a hint that the expected encoding must be UTF-8 compatible.
+
+### Fixes
+
+- Updated keysyms using latest [xorgproto] (commit: `cd33097fc779f280925c6d6bbfbd5150f93ca5bc`):
+
+  For the sake of compatibility, this reintroduces some deleted keysyms and
+  postpones the effective deprecation of others that landed in xkbcommon 1.6.0.
+
+  - Additions (reverted removal):
+
+    - `XKB_KEY_dead_lowline`
+    - `XKB_KEY_dead_aboveverticalline`
+    - `XKB_KEY_dead_belowverticalline`
+    - `XKB_KEY_dead_longsolidusoverlay`
+
+  - The following keysyms names remain deprecated, but are set again (i.e. as
+    before xkbcommon 1.6.0) as the reference names for their respective keysyms,
+    in order to ensure the transition to the newer names that replace them. This
+    affects functions such as `xkb_keymap_key_get_name` and `xkb_keymap_get_as_string`.
+
+    - `XKB_KEY_masculine`: is deprecated in favor of `XKB_KEY_ordmasculine`
+    - `XKB_KEY_guillemotleft`: is deprecated in favor of `XKB_KEY_guillemetleft`
+    - `XKB_KEY_guillemotright`: is deprecated in favor of `XKB_KEY_guillemetright`
+    - `XKB_KEY_dead_small_schwa`: is deprecated in favor of `XKB_KEY_dead_schwa`
+    - `XKB_KEY_dead_capital_schwa`: is deprecated in favor of `XKB_KEY_dead_SCHWA`
+
+  Relevant upstream merge requests: [xorgproto-83], [xorgproto-84].
+
+[xorgproto-83]: https://gitlab.freedesktop.org/xorg/proto/xorgproto/-/merge_requests/83
+[xorgproto-84]: https://gitlab.freedesktop.org/xorg/proto/xorgproto/-/merge_requests/84
+
+- Keysyms: Fixed case mapping for 3 Latin 1 keysyms:
+
+  - `XKB_KEY_ydiaeresis`
+  - `XKB_KEY_mu`
+  - `XKB_KEY_ssharp`
+
+- Keysyms: Fixed `xkb_keysym_is_modifier` to detect also the following keysyms:
+
+  - `XKB_KEY_ISO_Level5_Shift`
+  - `XKB_KEY_ISO_Level5_Latch`
+  - `XKB_KEY_ISO_Level5_Lock`
+
+- Prevent recursive includes of keymap components.
+
+- Compose: Fixed a segfault with `xkb_compose_table_iterator_next` when used on an
+  empty table.
+
+- Compose: Added check to ensure to open only regular files, not e.g. directories.
+
+- Registry: Updated the DTD and always parse the “popularity” attribute.
+
+- Fixed parsing of empty symbol declaration. The following erroneous syntax is now
+  correctly rejected: `xkb_symbols { key <SPC> {, [Space] }; };`.
+
+- Fixed a few memory leaks.
+
+Documentation
+-------------
+
+- Added documentation for the keymap format about multiple keysyms per level.
+- Added documentation to `XkbToControl`.
+
+Tools
+---
+
+### New
+
+- `xkbcli compile-compose`: added new CLI utility to test Compose files.
+- `xkbcli interactive-x11`: added support for Compose
+- `xkbcli interactive-wayland`: added support for Compose
+
+### Fixes
+
+- `xkbcli interactive-evdev`: fixed help.
```

Comment:
fwiw, tiny fixes like this don't IMO need to be mentioned in the release notes since you'd mostly expect every release to have fixes like this all over the place. No need to remove though, this is just a fwiw.

Reply:
Agreed. Git history has all the details.